### DISCUSSION TITLE
ucci set options

### DISCUSSION THREAD

**Discussion by wawahuy** [fairy-stockfish/Fairy-Stockfish#367] [Q&A]
I use CPU AMD Ryzen Threadripper 3970X, windows 10 21H1, play xiangqi
I use ucci command, I don't know what to set option but how to get the best performance and elo, I have some set options:
```
setoption Threads 32
setoption Hash 1024
setoption EvalFile D:\Data\Test\nnue.nnue
```

When I set the hash bigger, the slower it plays, I can go to depth 21 - 24. Sometimes I play with 1 thread and it's even faster than 32. The file nnue I downloaded at https://tests.stockfishchess.org/nns.
It would be great if someone could guide me to a complete option

**Comment by ianfab** [comment]
Yes, the number of threads and the amount of hash are the main relevant parameters for playing strength. As mentioned in https://github.com/ianfab/Fairy-Stockfish/wiki/FAQ#what-are-the-correct-settings-for-best-performance-on-my-machine, the optimal settings for hash depend a bit on the conditions. However, time to depth usually is not a good measure for playing strength. E.g., when using more threads they search at same/similar depths in parallel, which overall makes the search result better even if it does not reach a higher depth than with a single thread.

The NNUE files from https://tests.stockfishchess.org/nns are only for standard chess, they can not be used for Xiangqi. An overview of NNUE networks available for variants is available at https://docs.google.com/spreadsheets/d/1fgGvBKleUOI1wZZbiNhpT98qhQ7mYuQ5kar1lTQ9g3w/edit?usp=sharing. For Xiangqi I did not publish one yet.
